client/src/components/events-components/EventList.tsx

```diff
@@ -1,13 +1,37 @@
+import { useEffect, useState } from "react";
 import EventCard from "./EventCard";
+import { getAllEvents } from "@/utils/eventService";
+
+interface Event {
+  created_at: string;
+  description: string;
+  eventImage: string;
+  eventTime: string;
+  location: string;
+  latitude: number;
+  longitude: number;
+  organizerId: number;
+  id: number;
+  sport: string;
+  updated_at: string;
+}
 
 const EventList = () => {
-  const events = [{ name: "john" }, { name: "Hernyr" }];
+  const [events, setEvents] = useState([]);
+
+  useEffect(() => {
+    const fetchEvents = async () => {
+      const retrievedEvents = await getAllEvents();
+      setEvents(retrievedEvents);
+    };
+    fetchEvents();
+  }, []);
```

Comment:
```suggestion
  useEffect(() => {
      const retrievedEvents = await getAllEvents();
      setEvents(retrievedEvents);
  }, []);
```

the best practice is to decouple presentation from data, i.e. fetch the data outside of the UI component, which make `EventList` a pure UI component that only received and displays data.